File: AndroidWithJava/SharedPreference/app/src/main/java/com/example/sharedpreference/MainActivity.java
package com.example.sharedpreference;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.os.Handler;
import android.text.BoringLayout;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                SharedPreferences pref = getSharedPreferences("login", MODE_PRIVATE);
                Boolean check= pref.getBoolean("flag",false);
                Intent inext;
                if(check){
                   inext=new Intent(MainActivity.this,HomeActivity.class);

                }
                else {
                    inext=new Intent(MainActivity.this,LoginActivity.class);
                }
                startActivity(inext);
                finish();
            }
        }, 2000);
    }
}

File: AndroidWithKotlin/test/app/src/main/java/com/example/test/MainActivity.kt
package com.example.test

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import butterknife.ButterKnife
import com.example.test.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.myTextVieww.text = "Hello demo!"
    }



}